components/cli/internal/test/kubecli.go

```diff
@@ -18,16 +18,20 @@
 
 package test
 
-import "github.com/cellery-io/sdk/components/cli/kubernetes"
+import (
+	"github.com/cellery-io/sdk/components/cli/kubernetes"
+)
 
 type MockKubeCli struct {
 	cells kubernetes.Cells
+	composites kubernetes.Composites
 }
 
 // NewMockKubeCli returns a mock cli for the cli.KubeCli interface.
-func NewMockKubeCli(cells kubernetes.Cells) *MockKubeCli {
+func NewMockKubeCli(cells kubernetes.Cells, composites kubernetes.Composites) *MockKubeCli {
```

Comment:
We can improve this MockKubeCli creation function to accept Cells, Composites, Components using a chaining pattern:
NewMockKubeCli().WithCells(cells).WithComposites(composites).. etc.

Reply:
Improved MockKubeCli creations with functional options. This is the approach used by docker cli for unit testing as well.
changes updated in https://github.com/wso2-cellery/sdk/pull/874